vmm/src/memory_manager.rs

```diff
@@ -387,6 +401,14 @@ impl MemoryManager {
             let mut total_ram_size: u64 = 0;
             for zone in zones.iter() {
                 total_ram_size += zone.size;
+
+                if zone.shared && zone.host_numa_node.is_some() {
+                    error!(
+                        "Invalid to set host NUMA policy for a memory zone \
+                        mapped as 'shared'"
+                    );
+                    return Err(Error::InvalidSharedMemoryZoneWithHostNuma);
+                }
```

Comment:
Sorry to confuse you before. Actually mbind can bind vma  to shared region, unless the vma is mapping of regular file(e.g. ext4 file).

/dev/shm is tmpfs, and it's page fault hanlder is [shmem_fault](https://elixir.bootlin.com/linux/latest/source/mm/shmem.c#L1968), it will call [alloc_page_vma(gfp, &pvma, 0);](https://elixir.bootlin.com/linux/latest/source/mm/shmem.c#L1495) this function will use vma to allocate page, and the page can be apply with mempolicy that set using mbind.

and page fault handler of regular file use  __alloc_pages_node(n, gfp, 0); to allocate pages. that have no vma info.

Addition, vm_operations_struct of regular file have NOT `.set_policy`, and shm has [`.set_policy`](https://elixir.bootlin.com/linux/latest/source/mm/shmem.c#L3823)

Reply:
Ok I understand. I can change this check to make it happen when I know the type of file that is being created, and this would let you use mbind with `virtio-fs`, right?

Reply:
Ok I took care of this issue by adding one more commit which checks if the memory zone is backed by a file or not.